### DISCUSSION TITLE
Typing the store doesn't block invalid properties

### DISCUSSION THREAD

**Discussion by AndreiTS** [pmndrs/zustand#1925] [Bug report] [answered]
Why the property "hi" is allowed here and TS doesn't error?
![image](https://github.com/pmndrs/zustand/assets/13489837/54204a4a-6b23-4380-ba56-f735078f56ce)

workaround:
![image](https://github.com/pmndrs/zustand/assets/13489837/5407e8e3-ebae-4fc8-a284-29ac894f618f)

**Comment by dai-shi** [comment, marked as answer]
It's not invalid.
Your workaround seems good.

That's how TS works: https://tsplay.dev/N9QY1N
<img width="512" alt="image" src="https://github.com/pmndrs/zustand/assets/490574/983381ba-548e-493b-83de-5341710e146b">
